### DISCUSSION TITLE
MUlechain connector

### DISCUSSION THREAD

**Discussion by imohiuddin1** [MuleSoft-AI-Chain-Project/mule-ai-chain-connector#49] [Q&A] [answered]
Mulechain AI LLM Configuration is not populating any of the values in LLM Type drop down, Config type and other values.
I am assuming connector has predfeined in the drop down and we just need to select but nothing is showing in the drop down.

![image](https://github.com/user-attachments/assets/ad7220af-5d21-466a-9194-2b6855bc9f82)

**Comment by amirkhan-ak-sf** [comment]
@imohiuddin1  thanks for reporting this. I believe you are on `0.2.0 `version, correct? 
If so, I would suggest you download the latest `0.3.0` version from the master branch. It is a known issue, which got fixed in the latest push on master.

**Comment by imohiuddin1** [comment]
Thanks. I just cloned the new version and build failed.

**Reply by amirkhan-ak-sf** [reply]
Any logs? What failed?

**Comment by imohiuddin1** [comment]
[log.txt](https://github.com/user-attachments/files/16909595/log.txt)
Log is attached.

**Reply by imohiuddin1** [reply]
Here is the debug logs enabled.
[log.txt](https://github.com/user-attachments/files/16909778/log.txt)

**Reply by amirkhan-ak-sf** [reply, marked as answer]
@imohiuddin1 this is a known issue for windows. Please see this [thread](https://github.com/MuleSoft-AI-Chain-Project/mulesoft-ai-chain-connector/issues/33). Remove the maven-jdeps-plugin and jacoco-maven-plugin entries from the POM. and this should solve the issue.

**Reply by imohiuddin1** [reply]
Thanks. This resolved the issue and build was successful.
 The plugin still doesn't populates LLM drop down values. I see an error popup in studio saying "A studio service is timing out. Metadata resolution is taking longer than expected. Click Studio services inidcator to resolve."

I have changes the data sense properties and increased the time out, restarted the studio services but the error continues.

**Reply by imohiuddin1** [reply]
It worked and i was able to use chat. I will try other ones very soon.

**Reply by karthik928-web** [reply]
Hi @imohiuddin1, could you please let me know what are the changes required to fix the "studio service...." issue? So that the dropdown is populated?

Could you please post some details about the data sense properties and the timeout increase and any other changes you did to make it work?

**Reply by amirkhan-ak-sf** [reply]
@karthik928-web Remove the maven-jdeps-plugin and jacoco-maven-plugin from the connector pom.xml.

**Reply by karthik928-web** [reply]
@amirkhan-ak-sf thank you. Modifying the connector pom.xml solved the issue for the connector build. The build is successful. 

When I imported the connector as a dependency in my application project and create a connector config.....the metadata resolution seems to be failing...the dropdown doesn't get populated. I get the error "A studio service is timing out. Metadata resolution is taking longer than expected. Click Studio services inidcator to resolve."

looks like @imohiuddin1 has made that work as well.....is there any other configuration that I am missing?

**Reply by amirkhan-ak-sf** [reply]
@karthik928-web which version of the connector are you using? Try to enter "something" in the filepath of the configuration dialog and activate dataweave expressions (fx), then refresh the list.

**Reply by imohiuddin1** [reply]
I have moved to latest MuleChain connector and it worked.

**Comment by neha3112-hp** [comment]
Hi all while using chat prompt connector getting this error . I have tried with gpt-3.5 turbo and gpt-4o. Please suggest which version will work .
"dev.ai4j.openai4j.OpenAiHttpExcep tion: Only gpt-40-mini model is available for demonstration purposes. If you wish to use another model, please use your own OpenAI API key."

**Reply by amirkhan-ak-sf** [reply]
@neha3112-hp in order to use other models you have to provide your personal paid OpenAI API Key. demo only supports gpt-4o-mini.
